src/create-arrow-function.ts

```diff
@@ -0,0 +1,90 @@
+import { Modification, Update, NoModification } from "./modification";
+import { Position, Selection, Code, isFilled } from "./editor";
+import * as t from "./ast";
+
+export { createArrowFunction };
+export { Match, isMatch, CreateArrowFunction };
+
+function createArrowFunction(code: Code, selection: Selection): Modification {
+  let match: Match | undefined;
+  t.traverseCode(code, {
+    CallExpression(path) {
+      if (!selection.isInsidePath(path)) return;
+      if (!isMatch(path)) return;
+      if (t.isDeclared(path.node.callee, path)) return;
+
+      match = path;
+    }
+  });
+
+  if (!match) {
+    return new NoModification();
+  }
+
+  return new CreateArrowFunction(match, code);
+}
+
+function isMatch(path: t.NodePath<t.CallExpression>): path is Match {
+  return t.isIdentifier(path.node.callee);
+}
+
+type Match = t.NodePath<
+  t.Selectable<t.CallExpression> & { callee: t.Identifier }
+>;
+
+class CreateArrowFunction implements Modification {
```

Comment:
I think we can get rid of most duplication if we were extending `CreateFunction`. This is really a simple variant of it at this point.